typed_python/SerializationContext.py

```diff
@@ -41,24 +43,42 @@ def __init__(self, nameToObject=None):
         self.nameToObject = nameToObject or {}
         self.objToName = {}
 
-        #take the lexically lowest name, so that we're not dependent on ordering.
+        for k in self.nameToObject:
+            assert isinstance(k, str), (
+                "nameToObject keys must be strings (This one was not: {})"
+                .format(k)
+            )
+
+        assert '' not in self.nameToObject, (
+            "Empty object/type name not allowed: {}"
+            .format(self.nameToObject[''])
+        )
+
+        # take the lexically lowest name, so that we're not dependent on ordering.
         for k,v in self.nameToObject.items():
             if id(v) not in self.objToName or k < self.objToName[id(v)]:
                 self.objToName[id(v)] = k
 
         self.numpyCompressionEnabled = True
 
     def nameForObject(self, t):
-        if id(t) in _builtin_value_to_name:
-            return _builtin_value_to_name[id(t)]
+        ''' Return a name(string) for an input object t, or None if not found. '''
+        tid = id(t)
+        res = self.objToName.get(tid)
 
-        return self.objToName.get(id(t))
+        if res:
+            return res
+        else:
+            return _builtin_value_to_name.get(tid)
 
     def objectFromName(self, name):
-        if name in _builtin_name_to_value:
-            return _builtin_name_to_value[name]
+        ''' Return an object for an input name(string), or None if not found. '''
+        res = self.nameToObject.get(name)
 
-        return self.nameToObject.get(name)
+        if res:
```

Comment:
should be `res is None` or it will filter out objects that have boolean conversin to false

Reply:
yes! Good catch